soepy/simulate/simulate_auxiliary.py

```diff
@@ -101,21 +102,20 @@ def pyth_simulate(
         # Extract corresponding utilities
         current_log_wage_systematic = log_wage_systematic[idx]
         current_non_consumption_utilities = non_consumption_utilities[idx]
+        current_non_employment_benefits = non_employment_benefits[idx]
 
         current_wages = np.exp(
             current_log_wage_systematic.reshape(-1, 1)
             + draws_sim[period, current_states[:, 0]]
         )
-        current_wages[:, 0] = model_params.benefits
+        current_wages[:, 0] = current_non_employment_benefits
 
         # Calculate total values for all choices
         flow_utilities = np.full((current_states.shape[0], 3), np.nan)
 
         flow_utilities[:, :1] = (
```

Comment:
Please add a brief comment on what is going on here? You probably have very good reason for it, but it hard to grasp from an outside perspective.

Reply:
The reshape here is doing nothing else than transforming a row vector to a column vector.

Reply:
Can we avoid the need to do this by setting up the container differently right from the start?

Reply:
Here sure. Might result in need to reshape in other parts of the code. I will check if it really bothers you.

Reply:
Please create an issue for this so we revisit it at one of the next PRs